source/tutorials/dev-environment.md

```diff
@@ -28,67 +13,104 @@ app = Flask(__name__)
 
 @app.route("/")
 def hello():
-    return "Hello, Nix!"
+    return {
+        "message": "Hello, Nix!"
+    }
 
 def run():
-    app.run(host="0.0.0.0")
+    app.run(host="0.0.0.0", port=5000)
 
 if __name__ == "__main__":
     run()
 ```
 
-And a `setup.py` script:
+This is a simple Flask application which serves a JSON document with the message
+"Hello, Nix!".
+
+To declare the development environment, create a new file `shell.nix`: 
 
-```{code-block} python setup.py
-from setuptools import setup
+```{code-block} nix shell.nix
+{ pkgs ? import (fetchTarball "https://github.com/NixOS/nixpkgs/archive/eabc38219184cc3e04a974fe31857d8e0eac098d.tar.gz") {} }:
 
-setup(
-    name='myapp',
-    version='0.1',
-    py_modules=['myapp'],
-    entry_points={
-        'console_scripts': ['myapp = myapp:run']
-    },
-)
+pkgs.mkShell {
+  packages = [
+    (pkgs.python3.withPackages (ps: [
+      ps.flask
+    ]))
+
+    pkgs.curl
+    pkgs.jq
+  ];
+}
 ```
 
-Now build the package with:
+This creates a shell environment with `python3`, including the `flask` package.
+
+However, it also contains developer tools: [`curl`] (utility to perform web
+requests) and [`jq`] (utility to parse and format JSON documents). Both of them
+are not Python packages and they are not part of the Python ecosystem.
+
+If we went with Python's [virtualenv], it is not possible to have these
+utilities to be a part of the development environment without additional manual
+steps.
```

Comment:
```suggestion
If we went with Python's [virtualenv], it would not be possible to add these utilities
to the development environment without additional manual steps.
```